### ISSUE TITLE
Allow an option to generate 'true' and 'false' as valid values for booleans

### ISSUE THREAD

**Issue by filmil** [matejak/argbash#51]
Sometimes, I would like to use `true` and `false` for boolean flag values.  For example, some of the flags
would make their way into go programs.  argbash uses `on` and `off` as keywords to mean a flag is set or unset

Sadly these values are not immediately acceptable to say go programs -- though go acceps a number of different values, see here for details:
https://golang.org/pkg/flag/#pkg-overview

So I end up having a function to convert `off` and `on` into appropriate flags.  Any chance that argbash could have a generation option to configure these values?

The reason I report this to argbash instead of just rolling my own scripting is because, in general, I want to minimize the amount of arcane bash that I need to write.  If I could delegate part of this to an automated code generator, I'd be a happier camper.

Cheers!

**Comment by matejak** [comment]
To choose `on` instead of `true` was a deliberate decision - in shells, `true` is also a command with return value of 0. Although a string `true` would be OK, I wanted to avoid the confusion, and wanted to go for something that is as unambiguous as possible.
Please, consider `on` and `off` as part of the API. You can translate those values easily using dictionaries or associative arrays in the script or in the called program - fortunatelly, you have to do it only once.

**Comment by filmil** [comment]
Noted and thank you for explaining.  That said, I am not sure that I follow the line of reasoning about boolean flag values:
-  I would think that if someone is misusing something what is supposed to be a boolean flag value, they have a bigger issue than mixing the program `true` and a string `"true"`.
- Sometimes the program to be ran was not written by me, so would not accept an `on` value as "true".  Note that for example, go supports a few pairs of values (true and false; 1 and 0; T and F) which would all work.  I guess they could support "on" and "off" as well, but that's not likely to happen on a dime.
- The point of code generation is to minimize the amount of adaptation needed in addition to generated code.  So if I have to write an adapter to parse flags "more" than the parsing library already doesn, that *feels* like defeat.

But I do accept your explanation -- `argbash` doesn't need to work for everyone perfectly, and in the end it is possible to make a translation function, or even fork your own argbash.

**Comment by matejak** [comment]
You are right that confusing the string `true` for something that could be interpreted as not true is unlikely. However, [this doc snippet](https://github.com/kward/shflags/wiki/Documentation12x#boolean) devotes quite a lot of energy into explaining what true is and what isn't. I have realized that even small children in non-English-speaking countries know the meaning of on and off, so I went for that path, as I believe that it is more deeply rooted in our subconsciousness.

However thank you for your feedback, nothing is set in stone, and who knows how argbash 3.0 API will look like :-)
